2016-09/september-07.md

```diff
@@ -0,0 +1,87 @@
+	* We will have to investigate the `node.js` repo and see if there is anyone coming across this issue already with useful knowledge. Otherwise we will create an edge-case target so that this can be used the proper way. We will discuss this more in detail next meeting. 
+
+* **[#250](https://github.com/webpack/webpack/issues/250#issuecomment-199602293)**: 
+	* Johannes raised this issue because it is notable that some scaled builds take long to finish (even if it is 2-3 minutes). Johannes mentioned [`HappyPack`](https://github.com/amireh/happypack) is a common package that allows you to multithread the loader process. However there are struggles with the way that "hacky" loaders work that make `HappyPack` impossible for `awesome-typescript-loader` and `ts-loader` to implement together. 
+	* Sean describes this "hackyness" is when a loader accesses `compiler` properties from `this` such as `this._compiler` to execute plugin based functionality. Needs to be deprecated.
+	* Tobias mentions that maybe we can add functionality to the loader API to allow this to work. Sean explained that the reason these TS loaders need plugin access is because it needs to defer typechecking until the `compilation` is finished. 
+	* Tobias mentions that deferred execution could be a feature we need to add to the loader API. Could also be useful for image sprites, etc. 
+	* Johannes brings up that this is the same problem to `extract-text-plugin` (when you need to apply a plugin and a loader together).
+	* Tobias states that this could fix a great deal of "hackyness" from `extract-text-plugin` and move it to a loader. 
+	* Sean suggested that the deferred execution be expressed by unwrapping a promise. (
+	The loader function itself would return the promise. ) Would be minimal additional effort and easy for developers to understand.
+	* Tobias mentioned that tool that would be moved and use the new deferred execution API [(doesn't exist yet but suggested idea) `image-sprite-loader` and the `seperate-css-loader`] would need to include an `invalidate` method to issue recompilation from the callback. It would also need to include the following: 	
+		* errors/warnings
+		* invalidating => recompilation w/ new data
+		* emitting files
+		* access to the current chunk info (edited)
+	* additionally noted it would fall after optimization and before id creation (in the compilation step)
+	* This would also allow ts-loader and awesome-typescript-loader to execute typechecking in the resolve promise. (allowing tools like HappyPack to execute loaders parallel/multithreaded)
+	* Johannes mentions that it would be important that we get feedback from loader authors to see if this would help solve some of their issues (including HappyPack).
+	* Tobias described, (once implementing this feature [based upon the comments in the linked issue]) that a `disk-cache-loader` would look similar to having the following.
```

Comment:
One potential issue is with plugins that use an external config; e.g. eslint, babel (with .babelrc), ts (tsconfig.json is pretty much required). If the config changes, the output will most likely change too, but a naive approach would only notice it if it's part of the query.

Other loaders have the (much more convenient) option of reading their config from the webpack config object itself, like postcss-loader. As the config option may be a function, it can be harder to cache (in fact I use that property to use a different postcss syntax plugin per file type).

Perhaps a caching API could be provided, where the plugin itself can declare the state it is going to use to execute its load?

This would still leave funny cases such as browserslist files.